pkg/haprovider/haprovider.go

```diff
@@ -123,6 +122,45 @@ func (r *HAProvider) createService(
 	return service, err
 }
 
+func (r *HAProvider) getServiceAnnotation(ctx context.Context, cluster *clusterv1.Cluster) map[string]string {
+	serviceAnnotation := map[string]string{
+		akoov1alpha1.HAServiceAnnotationsKey:  "true",
+		akoov1alpha1.TKGClusterNameLabel:      cluster.Name,
+		akoov1alpha1.TKGClusterNameSpaceLabel: cluster.Namespace,
+	}
+
+	aviInfraSetting := &akov1alpha1.AviInfraSetting{}
+
+	// TODO: check a cluster should be managed by only one adc
+	adcForCluster, err := handlers.GetADCForCluster(ctx, cluster, r.log, r.Client)
+	if err != nil {
+		return serviceAnnotation
+	}
+
+	if len(adcForCluster) == 0 {
+		r.log.Error(err, "Failed to get akoDeploymentConfig for current cluster")
```

Comment:
If there is no adc for the current cluster, I don't think we need to throw an error, just having it logged should be good.

Reply:
Make sense to me!